jsonforms-tooling-common/src/index.ts

```diff
@@ -111,20 +113,32 @@ export const showPreview = async (editorInstance: any, firstSchemafileUri: any,
   } else {
     // If the user called this function by doing a right click on a json file, we need to know which schema file that was
     try {
-      uiSchemaOrSchema = await editorInstance.window.showQuickPick(['Schema', 'UI Schema'], editorInstance.QuickPickOptions = {
-        canSelectMany: false,
-        placeHolder: 'Was that the schema or the UI schema file?'
-      });
-      if (uiSchemaOrSchema === 'UI Schema') {
+      const schemaContent = await readFileWithPromise(firstSchemafileUri, 'utf8');
+      const parsedSchemaContent = JSON.parse(schemaContent);
+      const validUiSchema = await validateUiSchema(parsedSchemaContent);
+      if (validUiSchema) {
+        uiSchemaOrSchema = 'UI Schema';
+        otherSchema = 'Schema';
         selectSecondSchema = 'Select Schema';
         selectSecondErrorMessage = 'Please select a schema file';
       }
     } catch (err) {
-      showMessage('Please select the schema type', 'err');
+      showMessage(err.message, 'err');
       return;
     }
   }
   // In both situations we still need the second schema file
+  // First we'll tell the user what to do next
+  try {
+    await editorInstance.window.showQuickPick(['Next'], editorInstance.QuickPickOptions = {
+      canSelectMany: false,
+      placeHolder: `It seems, that you selected the ${uiSchemaOrSchema}. Now  please select the ${otherSchema} file.`
```

Comment:
Rephrase: You've selected a ${uiSchemaOrSchema} file. Continue by selecting a ${otherSchema} file.